src/smartcontracts/smartContract.local.net.spec.ts

```diff
@@ -325,4 +330,51 @@ describe("test on local testnet", function () {
         queryResponse = await provider.queryContract(query);
         assert.equal(decodeUnsignedNumber(queryResponse.getReturnDataParts()[0]), 0);
     });
+
+    it("counter: should deploy and call using the SmartContractFactory", async function () {
+        this.timeout(80000);
+
+        TransactionWatcher.DefaultPollingInterval = 5000;
+        TransactionWatcher.DefaultTimeout = 50000;
+
+        const network = await provider.getNetworkConfig();
+        await alice.sync(provider);
+
+        const transactionComputer = new TransactionComputer();
+        const config = new TransactionsFactoryConfig(network.ChainID);
+        const factory = new SmartContractTransactionsFactory({ config: config, tokenComputer: new TokenComputer() });
+
+        let bytecode = await promises.readFile("src/testdata/counter.wasm");
+
+        const deployTransaction = factory.createTransactionForDeploy({
+            sender: alice.address,
+            bytecode: bytecode,
+            gasLimit: 3000000n,
+        });
+        deployTransaction.nonce = BigInt(alice.account.nonce.valueOf());
```

Comment:
A bit verbose, but good at this moment. In the cookbook I think we will not mention `AccountNonce` class anymore - left for the client code to hold & handle the nonce.